test/http-timeout.test.js

```diff
@@ -52,6 +52,31 @@ async function main () {
   }
 
   t.fail()
-}
+})
 
-main()
+test('http sse removes timeout test', async (t) => {
+  const target = Fastify()
+  t.teardown(target.close.bind(target))
+
+  target.get('/', (request, reply) => {
+    t.pass('request arrives')
+
+    reply.header('content-type', 'text/event-stream').status(200).send('hello world')
+  })
```

Comment:
I admit I was struggling a little figuring out the best way to test these changes. The way this works is it more tests the code via the coverage: without the new tests, the coverage lowers.

If it's preferred to instead specifically test that `req.setTimeout(0)` is being called, for sure can let me know